server/health.go

```diff
@@ -31,7 +40,21 @@ func (s *Server) checkCRIHealth(ctx context.Context, timeout time.Duration) erro
 		return errors.New("runtime conditions are nil")
 	}
 
+	s.cniPluginReadinessCheck(ctx)
+
 	for _, c := range response.GetStatus().GetConditions() {
+		if c.GetType() == "NetworkReady" {
+			if !cniPluginInitialized.Load() {
+				log.Warnf(ctx, "CNI plugin not yet initialized. Ignoring NetworkReady status: %v, message: %s, reason: %s", c.GetStatus(), c.GetMessage(), c.GetReason())
+				continue
```

Comment:
We don't have to check this when the `NetworkReady` condition is true, I guess?

Reply:
Actually, we need to check this when `NetworkReady` is true. If the CNI plugin isn't initialized, CRI-O may fail to send heartbeats, causing systemd to send a SIGABRT. This check ensures CRI-O isn't marked unhealthy prematurely and helps prevent failures by confirming the network plugin is fully ready.